docs/samples/bentoml/README.md

```diff
@@ -0,0 +1,171 @@
+# Predict on an InferenceService using BentoML
+
+[BentoML](https://bentoml.org) is an open-source platform for high-performance ML model
+serving. It makes building production API endpoint for your ML model easy and supports all
+major machine learning training frameworks, including Tensorflow, Keras, PyTorch, XGBoost,
+scikit-learn and etc.
+
+BentoML comes with a high-performance API model server with adaptive micro-batching support,
+which achieves the advantage of batch processing in online serving. It also provides model
+management and model deployment functionality, giving ML teams an end-to-end model serving
+workflow, with DevOps best practices baked in.
+
+In this guide, we will demo serving a scikit-learn based iris classifier model with BentoML
+as InferenceService on a Kubernetes cluster with KFServing installed. The same deployment
+steps is also applicable for models trained with other machine learning frameworks, see
+more BentoML examples [here](https://docs.bentoml.org/en/latest/examples.html).
+
+## Deploy a custom InferenceService
+
+### Setup
+
+* Your ~/.kube/config should point to a cluster with KFServing installed.
+* Your cluster's Istio Ingress gateway must be network accessible.
+* Docker and Docker hub must be properly configured on your local system
+* Python 3.6 or above
+  * Install required packages `bentoml` and `scikit-learn` on your local system:
+
+    ```shell
+    pip install bentoml scikit-learn
+    ```
+
+### Build API model server using BentoML
+
+The followwing code defines an API model server that requires a `scikit-learn` model, and
+asks BentoML to figure out the required PyPI pip packages automatically. It also defined
+an API, which is the entry point for accessing this prediction service. And the API is expecting a `pandas.DataFrame` object as its input data.
+
+```python
+# iris_classifier.py
+from bentoml import env, artifacts, api, BentoService
+from bentoml.handlers import DataframeHandler
+from bentoml.artifact import SklearnModelArtifact
+
+@env(auto_pip_dependencies=True)
+@artifacts([SklearnModelArtifact('model')])
+class IrisClassifier(BentoService):
+
+    @api(DataframeHandler)
+    def predict(self, df):
+        return self.artifacts.model.predict(df)
+```
+
+The following code to create a BentoService SavedBundle with the trained an iris
```

Comment:
The following code trains a classifier model and serve it with the `IrisClassifier` defined above:

```
code
```

After saving the BentoService instance, you can now start a REST API server with the model trained and test it locally:

```python
bentoml serve IrisClassifier:latest
```
```bash
curl -i \
  --header "Content-Type: application/json" \
  --request POST \
  --data '[[5.1, 3.5, 1.4, 0.2]]' \
  http://localhost:5000/predict
```